app/options.go

```diff
@@ -127,9 +125,13 @@ func StartSearchEngine(s *Server) error {
 	return nil
 }
 
+// SetLogger requires platform service to be initialized before calling.
+// If not, logger should be set after platform service are initialized.
```

Comment:
This will anyways not work right? The `SetLogger` option is only called during server initialization, and those are called here: https://github.com/mattermost/mattermost-server/pull/20831/files#diff-dfb9eb62d98e724b29fb7874fa20c0f84ad0aeadd9234d8a9924ee379c671bbfL222.

If this will always fail, there's no point in keeping this. We should just remove.

If it would have worked, we could have used it in https://github.com/mattermost/enterprise/pull/1263/files#diff-062fe75c09a2abfa339ec4077ee29643cdb0e859b6e70addf5cac552716051ccR86.

Reply:
It will work only if `Config` option is set upfront. In that case we already initialize platform service. But otherwise yes, this options are called before platform initialized. I think the best approach here is to move options to platform slowly. But in that case the callers of `NewServer` would need to think about platform service creation. Maybe we can add a better logic here by reflecting server options to platform by treating them (eg. `Config`, `SetLogger`, `StartMetrics`) differently than others.

Reply:
Ah yes, if `Config` is set, then platform gets initialized. Ok, so then you can use the `SetLogger` option in https://github.com/mattermost/enterprise/pull/1263/files#diff-062fe75c09a2abfa339ec4077ee29643cdb0e859b6e70addf5cac552716051ccR86 as well right?